src/components/Country.js

```diff
@@ -0,0 +1,80 @@
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import fetchCountries from '../redux/countries/fetchCountries';
+import filterData from './filterData';
+import './Country.css';
+
+const Country = () => {
+  const dispatch = useDispatch();
+  const countriesLoaded = useSelector(
+    (state) => state.countries.countries.length > 0,
+  );
+
+  useEffect(() => {
+    if (!countriesLoaded) {
+      dispatch(fetchCountries());
+    }
+  }, [dispatch, countriesLoaded]);
+
+  const data = useSelector((state) => state.countries.countries);
+  let filteredCountries = filterData(data);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleSearchInputChange = (event) => {
+    const inputValue = event.target.value;
+    const formattedValue = inputValue
+      .toLowerCase()
+      .replace(/(?:^|\s)\w/g, (match) => match.toUpperCase());
+    setSearchQuery(formattedValue);
+  };
+
+  filteredCountries = data.filter(
+    (country) => country.name.toLowerCase().includes(searchQuery.toLowerCase()),
+  );
+
+  return (
+    <div>
+      <p className="search">
+        <input
+          type="text"
+          placeholder="Search events"
+          value={searchQuery}
+          onChange={handleSearchInputChange}
+        />
+      </p>
+      <div className="country-container">
+        {filteredCountries.map((country, index) => {
+          let boxClass = '';
+
+          if (index === 1) {
+            boxClass = 'light';
+          } else if (index % 4 === 1 || index % 4 === 2) {
+            boxClass = 'dark';
+          } else {
+            boxClass = 'even';
+          }
+          return (
+            <Link to={`/country/${encodeURIComponent(country.name)}`} key={country.name} style={{ textDecoration: 'none' }}>
+              <div className={`country-box ${boxClass}`} key={country.name}>
+                <h2 className="country-name">{country.name}</h2>
+                <img
+                  className="flag"
+                  src={country.flag}
+                  alt={`Flag of ${country.name}`}
+                />
+                <div className="population">
+                  Population:
+                  {' '}
+                  {country.population}
+                </div>
+              </div>
+            </Link>
```

Comment:
- [ ] Kindly try as much as you can to match the given design, this means that the colors (background, text color, fonts) should be the same as the ones from [Nelson sakwa](https://www.behance.net/sakwadesignstudio).


![cap1](https://github.com/tjay1760/Country-Metrics/assets/53124548/b686df66-12a6-40bc-93f9-0e9733456850)


![cap2](https://github.com/tjay1760/Country-Metrics/assets/53124548/a0762953-a6c7-48f1-8223-3752696d7ced)


![yours](https://github.com/tjay1760/Country-Metrics/assets/53124548/5832e790-dbb9-4e97-855a-6abb127a7957)